packages/realm-react/src/RealmProvider.tsx

```diff
@@ -27,6 +27,10 @@ type PartialRealmConfiguration = Omit<Partial<Realm.Configuration>, "sync"> & {
   sync?: Partial<Realm.SyncConfiguration>;
 };
 
+export type RealmFallbackComponent = React.ComponentType<{
+  progress: number | null;
```

Comment:
`null` is normally used for the absence of an object, perhaps `undefined` would be better here? Or just some sane number like `0` if the progress is unknown?